people-and-planet-ai/land-cover-classification/create_dataset.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This creates training and validation datasets for the model.
+
+As inputs it takes a Sentinel-2 image consisting of 13 bands.
+Each band contains data for a specific range of the electromagnetic spectrum.
+    https://developers.google.com/earth-engine/datasets/catalog/COPERNICUS_S2_HARMONIZED
+
+As outputs it returns the probabilities of each classification for every pixel.
+The land cover labels for the training dataset come from the ESA WorldCover.
+    https://developers.google.com/earth-engine/datasets/catalog/ESA_WorldCover_v100
+"""
+
+from typing import Iterable, List, Optional, Tuple
+
+import apache_beam as beam
+from apache_beam.options.pipeline_options import PipelineOptions
+import ee
+import numpy as np
+
+from serving import data
+
+# Default values.
+POINTS_PER_CLASS = 100
+PATCH_SIZE = 128
+MAX_REQUESTS = 20  # default EE request quota
+
+# Type aliases.
+Point = Tuple[float, float]  # (lon, lat)
+Polygon = List[Point]
+Example = Tuple[np.ndarray, np.ndarray]
+
+# Simplified polygons covering most land areas in the world.
+WORLD_POLYGONS = [
+    # Americas
+    [(-33.0, -7.0), (-55.0, 53.0), (-166.0, 65.0), (-68.0, -56.0)],
+    # Africa, Asia, Europe
+    [
+        (74.0, 71.0),
+        (166.0, 55.0),
+        (115.0, -11.0),
+        (74.0, -4.0),
+        (20.0, -38.0),
+        (-29.0, 25.0),
+    ],
+    # Australia
+    [(170.0, -47.0), (179.0, -37.0), (167.0, -12.0), (128.0, 17.0), (106.0, -29.0)],
+]
+
+
+def sample_points(
+    seed: int, polygons: List[Polygon], points_per_class: int, scale: int = 1000
+) -> Iterable[Point]:
```

Comment:
Kind of starting to get hard to read

```suggestion
def sample_points(
    seed: int,
    polygons: list[Polygon],
    points_per_class: int,
    scale: int = 1000,
) -> Iterable[Point]:
```